lib/collections/companies.js

```diff
@@ -0,0 +1,146 @@
+const dataplug = require('@dataplug/dataplug');
+const config = require('../config');
+const factory = require('../factory');
+
+const schema = {
+  type: 'object',
+  definitions: {
+    'source-type': require('../types/source-type'),
+    'company-property': {
+      type: 'object',
+      properties: {
+        value: {
+          type: 'string'
+        },
+        timestamp: {
+          type: 'integer'
+        },
+        source: {
+          $ref: '#/definitions/source-type'
+        },
+        sourceId: {
+          type: ['string', 'null']
+        },
+        versions: {
+          type: 'array',
+          items: {
+            $ref: '#/definitions/company-property-value-history'
+          }
+        }
+      },
+      additionalProperties: false
+    },
+    'company-property-value-history': {
+      type: 'object',
+      properties: {
+        name: {
+          type: 'string'
+        },
+        value: {
+          type: 'string'
+        },
+        timestamp: {
+          type: 'integer'
+        },
+        source: {
+          $ref: '#/definitions/source-type'
+        },
+        sourceId: {
+          type: ['string', 'null']
+        },
+        sourceVid: {
+          type: 'array',
+          items: {
+            type: 'integer'
+          }
+        }
+      },
+      additionalProperties: false
+    }
+  },
+  properties: {
+    portalId: {
+      type: 'integer'
+    },
+    companyId: {
+      type: 'integer'
+    },
+    isDeleted: {
+      type: 'boolean'
+    },
+    properties: {
+      type: 'object',
+      '^.*$': {
+        $ref: '#/definitions/company-property'
+      }
+    },
+  },
+  additionalProperties: false,
+  required: ['portalId', 'companyId']
+};
+
+const source = dataplug.source(
+  config.declaration.extended((declaration) => declaration
+    .parameters({
+      recentlyCreated: {
+        description: 'Fetch companies which were created in the last 30 days, or the 10k most recently created records sorted by the date the companies were created.'
+      },
+      recentlyUpdated: {
+        description: 'Fetch companies which were modified in the last 30 days, or the 10k most recently modified records sorted by the date the companies were most recently modified.'
+      }
+    })
+    .conflicts('recentlyCreated', 'recentlyCreated')
+    .parameters({
+      properties: {
+        description: 'Fetch companies with specific properties',
+        type: 'string'
+      },
+      propertiesWithHistory: {
```

Comment:
How will CLI react if I'll call without parameters at all?
How to retrieve all properties with history?

Reply:
If you will call without parameters then you will fetch all companies. By default, each company object without properties. We need to specify which properties we want to retrieve.

Reply:
That's not the desired behavior - it's not similar to contacts. By default, _all_ properties should be provided with history